test/integration/read-write-concern/write_concern.test.ts

```diff
@@ -161,11 +168,145 @@ describe('Write Concern', function () {
               { writeConcern: { w: 'majority' } }
             );
 
+            await sleep(1000);
+
             const err = await changes.next().catch(e => e);
             expect(err).to.not.be.instanceOf(Error);
           }
         });
       });
     });
   });
+
+  describe('fire-and-forget protocol', function () {
+    context('when writeConcern = 0 and OP_MSG is used', function () {
+      const writeOperations: { name: string; command: any; expectedReturnVal: any }[] = [
+        {
+          name: 'insertOne',
+          command: client => client.db('test').collection('test').insertOne({ a: 1 }),
+          expectedReturnVal: { acknowledged: false }
```

Comment:
I feel like the `expectedReturnVal` for these tests isn't necessary as it's just showing that we return custom values from these methods and only the assertion on the command suceeded event's reply is really needed here. Maybe we can remove these and the assertions on them?

Reply:
I'm fine leaving them as-is then. Cannot hurt.